tests/test_playbooks/organization_info.yml

```diff
@@ -0,0 +1,46 @@
+---
+- hosts: localhost
+  collections:
+    - theforeman.foreman
+  gather_facts: false
+  vars_files:
+    - vars/server.yml
+  tasks:
+    - include: tasks/organization.yml
+      vars:
+        organization_state: present
+
+- hosts: tests
+  collections:
+    - theforeman.foreman
+  gather_facts: false
+  vars_files:
+    - vars/server.yml
+  tasks:
+    - name: fetch organization info
+      organization_info:
+        username: "{{ foreman_username }}"
+        password: "{{ foreman_password }}"
+        server_url: "{{ foreman_server_url }}"
+        validate_certs: "{{ foreman_validate_certs }}"
+        organization: "Test Organization"
+        name: "Test Organization"
+      register: organization_info
+    - name: check organization details
+      assert:
+        that:
+          - organization_info['organization']['name'] == "Test Organization"
+
+    - name: search organization info
+      organization_info:
+        username: "{{ foreman_username }}"
+        password: "{{ foreman_password }}"
+        server_url: "{{ foreman_server_url }}"
+        validate_certs: "{{ foreman_validate_certs }}"
+        organization: "Test Organization"
+        search: "name ~ Test"
+      register: organization_info
+    - name: check organization details
+      assert:
+        that:
+          - organization_info['organizations'][0]['name'] == "Test Organization"
```

Comment:
Can we add more assertions on the output? I have the impression, that this second call to the module will produce by far less output (Or do we have a "detail" / "skip-detail" functionality? I do not remember.).

Reply:
> Can we add more assertions on the output? I have the impression, that this second call to the module will produce by far less output 

What assertions should I add?

> Or do we have a "detail" / "skip-detail" functionality?

TBH never heard about it

Reply:
The list call reports (taken from the recorded fixtures):
```
string: "{\n  \"total\": 2,\n  \"subtotal\": 1,\n  \"page\": 1,\n  \"per_page\":
        4294967296,\n  \"search\": \"name ~ Test\",\n  \"sort\": {\n    \"by\": null,\n
        \   \"order\": null\n  },\n  \"results\": [{\"label\":\"Test_Organization\",\"created_at\":\"2021-09-27
        12:22:17 UTC\",\"updated_at\":\"2021-09-27 12:22:21 UTC\",\"id\":5,\"name\":\"Test
        Organization\",\"title\":\"Test Organization\",\"description\":\"A test organization\"}]\n}\n"
```
The "detail"-call reports the organization as:
```
string: '{"label":"Test_Organization","redhat_repository_url":"https://cdn.redhat.com","system_purposes":{"addons":[],"roles":[],"usage":[],"support_level":[],"support_type":[]},"service_levels":[],"service_level":null,"select_all_types":[],"description":"A
        test organization","created_at":"2021-09-27 12:22:17 UTC","updated_at":"2021-09-27
        12:22:21 UTC","ancestry":null,"parent_id":null,"parent_name":null,"id":5,"name":"Test
        Organization","title":"Test Organization","users":[],"smart_proxies":[{"name":"centos7-katello-devel-stable.example.com","id":1,"url":"https://centos7-katello-devel-stable.example.com:9090","inherited":false}],"subnets":[],"compute_resources":[],"media":[],"ptables":[{"description":null,"os_family":"Suse","created_at":"2021-09-14
        23:40:12 UTC","updated_at":"2021-09-14 23:40:12 UTC","name":"AutoYaST entire
        SCSI disk","id":111,"inherited":false},{"description":null,"os_family":"Suse","created_at":"2021-09-14
        23:40:12 UTC","updated_at":"2021-09-14 23:40:12 UTC","name":"AutoYaST entire
        virtual disk","id":112,"inherited":false},{"description":null,"os_family":"Suse","created_at":"2021-09-14
        23:40:12 UTC","updated_at":"2021-09-14 23:40:12 UTC","name":"AutoYaST LVM","id":113,"inherited":false},{"description":null,"os_family":"Coreos","created_at":"2021-09-14
        23:40:12 UTC","updated_at":"2021-09-14 23:40:12 UTC","name":"CoreOS default
        fake","id":114,"inherited":false},{"description":null,"os_family":"Rancheros","created_at":"2021-09-14
        23:40:12 UTC","updated_at":"2021-09-14 23:40:12 UTC","name":"Empty","id":115,"inherited":false},{"description":null,"os_family":"Freebsd","created_at":"2021-09-14
        23:40:12 UTC","updated_at":"2021-09-14 23:40:12 UTC","name":"FreeBSD default
        fake","id":116,"inherited":false},{"description":null,"os_family":"Solaris","created_at":"2021-09-14
        23:40:12 UTC","updated_at":"2021-09-14 23:40:12 UTC","name":"Jumpstart default","id":117,"inherited":false},{"description":null,"os_family":"Solaris","created_at":"2021-09-14
        23:40:12 UTC","updated_at":"2021-09-14 23:40:12 UTC","name":"Jumpstart mirrored","id":118,"inherited":false},{"description":null,"os_family":"Junos","created_at":"2021-09-14
        23:40:12 UTC","updated_at":"2021-09-14 23:40:12 UTC","name":"Junos default
        fake","id":119,"inherited":false},{"description":null,"os_family":"Redhat","created_at":"2021-09-14
        23:40:12 UTC","updated_at":"2021-09-14 23:40:12 UTC","name":"Kickstart custom","id":120,"inherited":false},{"description":null,"os_family":"Redhat","created_at":"2021-09-14
        23:40:12 UTC","updated_at":"2021-09-14 23:40:12 UTC","name":"Kickstart default","id":121,"inherited":false},{"description":null,"os_family":"Redhat","created_at":"2021-09-14
        23:40:12 UTC","updated_at":"2021-09-14 23:40:12 UTC","name":"Kickstart default
        thin","id":122,"inherited":false},{"description":null,"os_family":"Redhat","created_at":"2021-09-14
        23:40:12 UTC","updated_at":"2021-09-14 23:40:12 UTC","name":"Kickstart dynamic","id":123,"inherited":false},{"description":null,"os_family":"NXOS","created_at":"2021-09-14
        23:40:12 UTC","updated_at":"2021-09-14 23:40:12 UTC","name":"NX-OS default
        fake","id":124,"inherited":false},{"description":null,"os_family":"Debian","created_at":"2021-09-14
        23:40:12 UTC","updated_at":"2021-09-14 23:40:12 UTC","name":"Preseed default","id":125,"inherited":false},{"description":null,"os_family":"Debian","created_at":"2021-09-14
        23:40:12 UTC","updated_at":"2021-09-14 23:40:12 UTC","name":"Preseed default
        LVM","id":126,"inherited":false},{"description":null,"os_family":"Windows","created_at":"2021-09-14
        23:40:12 UTC","updated_at":"2021-09-14 23:40:12 UTC","name":"Windows default
        GPT EFI partition table","id":128,"inherited":false},{"description":null,"os_family":"Windows","created_at":"2021-09-14
        23:40:12 UTC","updated_at":"2021-09-14 23:40:12 UTC","name":"Windows default
        partition table","id":127,"inherited":false},{"description":null,"os_family":"Xenserver","created_at":"2021-09-14
        23:40:12 UTC","updated_at":"2021-09-14 23:40:12 UTC","name":"XenServer default","id":129,"inherited":false}],"provisioning_templates":[{"id":44,"name":"Alterator
        default","template_kind_id":6,"template_kind_name":"provision","inherited":false},{"id":28,"name":"Alterator
        default finish","template_kind_id":7,"template_kind_name":"finish","inherited":false},{"id":10,"name":"Alterator
        default PXELinux","template_kind_id":2,"template_kind_name":"PXELinux","inherited":false},{"id":61,"name":"alterator_pkglist","template_kind_id":null,"template_kind_name":null,"inherited":false},{"id":62,"name":"ansible_provisioning_callback","template_kind_id":null,"template_kind_name":null,"inherited":false},{"id":63,"name":"ansible_tower_callback_script","template_kind_id":null,"template_kind_name":null,"inherited":false},{"id":64,"name":"ansible_tower_callback_service","template_kind_id":null,"template_kind_name":null,"inherited":false},{"id":45,"name":"Atomic
        Kickstart default","template_kind_id":6,"template_kind_name":"provision","inherited":false},{"id":46,"name":"AutoYaST
        default","template_kind_id":6,"template_kind_name":"provision","inherited":false},{"id":37,"name":"AutoYaST
        default iPXE","template_kind_id":5,"template_kind_name":"iPXE","inherited":false},{"id":11,"name":"AutoYaST
        default PXELinux","template_kind_id":2,"template_kind_name":"PXELinux","inherited":false},{"id":106,"name":"AutoYaST
        default user data","template_kind_id":9,"template_kind_name":"user_data","inherited":false},{"id":47,"name":"AutoYaST
        SLES default","template_kind_id":6,"template_kind_name":"provision","inherited":false},{"id":65,"name":"blacklist_kernel_modules","template_kind_id":null,"template_kind_name":null,"inherited":false},{"id":66,"name":"bmc_nic_setup","template_kind_id":null,"template_kind_name":null,"inherited":false},{"id":67,"name":"built","template_kind_id":null,"template_kind_name":null,"inherited":false},{"id":68,"name":"chef_client","template_kind_id":null,"template_kind_name":null,"inherited":false},{"id":27,"name":"CloudInit
        default","template_kind_id":12,"template_kind_name":"cloud-init","inherited":false},{"id":162,"name":"Content
        Proxy TODO","template_kind_id":8,"template_kind_name":"script","inherited":false},{"id":69,"name":"coreos_cloudconfig","template_kind_id":null,"template_kind_name":null,"inherited":false},{"id":48,"name":"CoreOS
        provision","template_kind_id":6,"template_kind_name":"provision","inherited":false},{"id":12,"name":"CoreOS
        PXELinux","template_kind_id":2,"template_kind_name":"PXELinux","inherited":false},{"id":70,"name":"create_users","template_kind_id":null,"template_kind_name":null,"inherited":false},{"id":71,"name":"csr_attributes.yaml","template_kind_id":null,"template_kind_name":null,"inherited":false},{"id":72,"name":"efibootmgr_netboot","template_kind_id":null,"template_kind_name":null,"inherited":false},{"id":73,"name":"eject_cdrom","template_kind_id":null,"template_kind_name":null,"inherited":false},{"id":74,"name":"epel","template_kind_id":null,"template_kind_name":null,"inherited":false},{"id":75,"name":"fips_packages","template_kind_id":null,"template_kind_name":null,"inherited":false},{"id":76,"name":"fix_hosts","template_kind_id":null,"template_kind_name":null,"inherited":false},{"id":29,"name":"FreeBSD
        (mfsBSD) finish","template_kind_id":7,"template_kind_name":"finish","inherited":false},{"id":49,"name":"FreeBSD
        (mfsBSD) provision","template_kind_id":6,"template_kind_name":"provision","inherited":false},{"id":13,"name":"FreeBSD
        (mfsBSD) PXELinux","template_kind_id":2,"template_kind_name":"PXELinux","inherited":false},{"id":77,"name":"freeipa_register","template_kind_id":null,"template_kind_name":null,"inherited":false},{"id":58,"name":"Global
        Registration","template_kind_id":13,"template_kind_name":"registration","inherited":false},{"id":161,"name":"Global
        Registration clone","template_kind_id":13,"template_kind_name":"registration","inherited":false},{"id":59,"name":"Grubby
        default","template_kind_id":8,"template_kind_name":"script","inherited":false},{"id":78,"name":"http_proxy","template_kind_id":null,"template_kind_name":null,"inherited":false},{"id":79,"name":"insights","template_kind_id":null,"template_kind_name":null,"inherited":false},{"id":38,"name":"iPXE
        default local boot","template_kind_id":5,"template_kind_name":"iPXE","inherited":false},{"id":39,"name":"iPXE
        global default","template_kind_id":5,"template_kind_name":"iPXE","inherited":false},{"id":40,"name":"iPXE
        intermediate script","template_kind_id":5,"template_kind_name":"iPXE","inherited":false},{"id":50,"name":"Jumpstart
        default","template_kind_id":6,"template_kind_name":"provision","inherited":false},{"id":30,"name":"Jumpstart
        default finish","template_kind_id":7,"template_kind_name":"finish","inherited":false},{"id":2,"name":"Jumpstart
        default PXEGrub","template_kind_id":3,"template_kind_name":"PXEGrub","inherited":false},{"id":31,"name":"Junos
        default finish","template_kind_id":7,"template_kind_name":"finish","inherited":false},{"id":51,"name":"Junos
        default SLAX","template_kind_id":6,"template_kind_name":"provision","inherited":false},{"id":26,"name":"Junos
        default ZTP config","template_kind_id":10,"template_kind_name":"ZTP","inherited":false},{"id":52,"name":"Kickstart
        default","template_kind_id":6,"template_kind_name":"provision","inherited":false},{"id":32,"name":"Kickstart
        default finish","template_kind_id":7,"template_kind_name":"finish","inherited":false},{"id":41,"name":"Kickstart
        default iPXE","template_kind_id":5,"template_kind_name":"iPXE","inherited":false},{"id":3,"name":"Kickstart
        default PXEGrub","template_kind_id":3,"template_kind_name":"PXEGrub","inherited":false},{"id":6,"name":"Kickstart
        default PXEGrub2","template_kind_id":4,"template_kind_name":"PXEGrub2","inherited":false},{"id":14,"name":"Kickstart
        default PXELinux","template_kind_id":2,"template_kind_name":"PXELinux","inherited":false},{"id":107,"name":"Kickstart
        default user data","template_kind_id":9,"template_kind_name":"user_data","inherited":false},{"id":81,"name":"kickstart_ifcfg_bonded_interface","template_kind_id":null,"template_kind_name":null,"inherited":false},{"id":80,"name":"kickstart_ifcfg_bond_interface","template_kind_id":null,"template_kind_name":null,"inherited":false},{"id":82,"name":"kickstart_ifcfg_generic_interface","template_kind_id":null,"template_kind_name":null,"inherited":false},{"id":83,"name":"kickstart_ifcfg_get_identifier_names","template_kind_id":null,"template_kind_name":null,"inherited":false},{"id":84,"name":"kickstart_kernel_options","template_kind_id":null,"template_kind_name":null,"inherited":false},{"id":85,"name":"kickstart_networking_setup","template_kind_id":null,"template_kind_name":null,"inherited":false},{"id":53,"name":"Kickstart
        oVirt-RHVH","template_kind_id":6,"template_kind_name":"provision","inherited":false},{"id":15,"name":"Kickstart
        oVirt-RHVH PXELinux","template_kind_id":2,"template_kind_name":"PXELinux","inherited":false},{"id":36,"name":"Linux
        host_init_config default","template_kind_id":1,"template_kind_name":"host_init_config","inherited":false},{"id":86,"name":"ntp","template_kind_id":null,"template_kind_name":null,"inherited":false},{"id":1,"name":"NX-OS
        default POAP setup","template_kind_id":11,"template_kind_name":"POAP","inherited":false},{"id":54,"name":"Preseed
        default","template_kind_id":6,"template_kind_name":"provision","inherited":false},{"id":33,"name":"Preseed
        default finish","template_kind_id":7,"template_kind_name":"finish","inherited":false},{"id":42,"name":"Preseed
        default iPXE","template_kind_id":5,"template_kind_name":"iPXE","inherited":false},{"id":7,"name":"Preseed
        default PXEGrub2","template_kind_id":4,"template_kind_name":"PXEGrub2","inherited":false},{"id":16,"name":"Preseed
        default PXELinux","template_kind_id":2,"template_kind_name":"PXELinux","inherited":false},{"id":108,"name":"Preseed
        default user data","template_kind_id":9,"template_kind_name":"user_data","inherited":false},{"id":87,"name":"preseed_networking_setup","template_kind_id":null,"template_kind_name":null,"inherited":false},{"id":88,"name":"puppet.conf","template_kind_id":null,"template_kind_name":null,"inherited":false},{"id":90,"name":"puppetlabs_repo","template_kind_id":null,"template_kind_name":null,"inherited":false},{"id":89,"name":"puppet_setup","template_kind_id":null,"template_kind_name":null,"inherited":false},{"id":91,"name":"pxegrub2_chainload","template_kind_id":null,"template_kind_name":null,"inherited":false},{"id":8,"name":"PXEGrub2
        default local boot","template_kind_id":4,"template_kind_name":"PXEGrub2","inherited":false},{"id":92,"name":"pxegrub2_discovery","template_kind_id":null,"template_kind_name":null,"inherited":false},{"id":9,"name":"PXEGrub2
        global default","template_kind_id":4,"template_kind_name":"PXEGrub2","inherited":false},{"id":93,"name":"pxegrub2_mac","template_kind_id":null,"template_kind_name":null,"inherited":false},{"id":94,"name":"pxegrub_chainload","template_kind_id":null,"template_kind_name":null,"inherited":false},{"id":4,"name":"PXEGrub
        default local boot","template_kind_id":3,"template_kind_name":"PXEGrub","inherited":false},{"id":95,"name":"pxegrub_discovery","template_kind_id":null,"template_kind_name":null,"inherited":false},{"id":5,"name":"PXEGrub
        global default","template_kind_id":3,"template_kind_name":"PXEGrub","inherited":false},{"id":17,"name":"PXELinux
        chain iPXE","template_kind_id":2,"template_kind_name":"PXELinux","inherited":false},{"id":18,"name":"PXELinux
        chain iPXE UNDI","template_kind_id":2,"template_kind_name":"PXELinux","inherited":false},{"id":96,"name":"pxelinux_chainload","template_kind_id":null,"template_kind_name":null,"inherited":false},{"id":19,"name":"PXELinux
        default local boot","template_kind_id":2,"template_kind_name":"PXELinux","inherited":false},{"id":20,"name":"PXELinux
        default memdisk","template_kind_id":2,"template_kind_name":"PXELinux","inherited":false},{"id":97,"name":"pxelinux_discovery","template_kind_id":null,"template_kind_name":null,"inherited":false},{"id":21,"name":"PXELinux
        global default","template_kind_id":2,"template_kind_name":"PXELinux","inherited":false},{"id":98,"name":"rancheros_cloudconfig","template_kind_id":null,"template_kind_name":null,"inherited":false},{"id":55,"name":"RancherOS
        provision","template_kind_id":6,"template_kind_name":"provision","inherited":false},{"id":22,"name":"RancherOS
        PXELinux","template_kind_id":2,"template_kind_name":"PXELinux","inherited":false},{"id":99,"name":"redhat_register","template_kind_id":null,"template_kind_name":null,"inherited":false},{"id":100,"name":"remote_execution_ssh_keys","template_kind_id":null,"template_kind_name":null,"inherited":false},{"id":101,"name":"saltstack_minion","template_kind_id":null,"template_kind_name":null,"inherited":false},{"id":102,"name":"saltstack_setup","template_kind_id":null,"template_kind_name":null,"inherited":false},{"id":103,"name":"schedule_reboot","template_kind_id":null,"template_kind_name":null,"inherited":false},{"id":109,"name":"UserData
        default","template_kind_id":9,"template_kind_name":"user_data","inherited":false},{"id":110,"name":"UserData
        open-vm-tools","template_kind_id":9,"template_kind_name":"user_data","inherited":false},{"id":23,"name":"WAIK
        default PXELinux","template_kind_id":2,"template_kind_name":"PXELinux","inherited":false},{"id":34,"name":"Windows
        default finish","template_kind_id":7,"template_kind_name":"finish","inherited":false},{"id":43,"name":"Windows
        default iPXE","template_kind_id":5,"template_kind_name":"iPXE","inherited":false},{"id":56,"name":"Windows
        default provision","template_kind_id":6,"template_kind_name":"provision","inherited":false},{"id":24,"name":"Windows
        default PXELinux","template_kind_id":2,"template_kind_name":"PXELinux","inherited":false},{"id":104,"name":"Windows
        network","template_kind_id":null,"template_kind_name":null,"inherited":false},{"id":60,"name":"Windows
        peSetup.cmd","template_kind_id":8,"template_kind_name":"script","inherited":false},{"id":57,"name":"XenServer
        default answerfile","template_kind_id":6,"template_kind_name":"provision","inherited":false},{"id":35,"name":"XenServer
        default finish","template_kind_id":7,"template_kind_name":"finish","inherited":false},{"id":25,"name":"XenServer
        default PXELinux","template_kind_id":2,"template_kind_name":"PXELinux","inherited":false},{"id":105,"name":"yum_proxy","template_kind_id":null,"template_kind_name":null,"inherited":false}],"domains":[],"realms":[],"hostgroups":[],"locations":[{"id":2,"name":"Default
        Location","title":"Default Location","description":null}],"hosts_count":0,"parameters":[],"deprecations":{"environments":"Environments
        got deprecated from this endpoint."},"simple_content_access":false,"owner_details":{"id":"ff8080817c26839d017c2734ba640009","key":"Test_Organization","created":"2021-09-27T12:22:19+0000","updated":"2021-09-27T12:22:19+0000","lastRefreshed":null,"virt_who":false,"upstreamConsumer":null},"default_content_view_id":4,"library_id":4}'
```

`parent_name` sticks out to me.

Reply:
Just to add a bit context (or confuse readers, depending on the view): we have an old `ressource_info` module, which allows (via a special parameter) to pull in *all* details (that are visible via a `show`, but not an `index` request).

the new, dedicated modules don't have this, and so far it was by design (if anyone needs to search for a long list, they can get the IDs and then re-run the module).

Reply:
We discussed, that we want to see one field asserted (maybe 'parent_name`) on the "detail" test case that will not appear on the list / fuzzy search test case. And this is good to go.

Reply:
Tests updated, added assertions for `parent_name`